t/lib/OpenQA/Test/Utils.pm

```diff
@@ -381,4 +379,11 @@ sub client_output {
     return $out;
 }
 
+sub shared_hash {
+    my $hash = shift;
+    state $file = do { my $f = tempfile; lock_store {}, $f->to_string; $f };
+    return lock_retrieve $file->to_string unless $hash;
+    lock_store $hash, $file->to_string;
+}
```

Comment:
I don't love this, but it was the most minimalistic solution i could think of for read/write sharing a data structure between two processes.